packages/utils/src/promise.ts

```diff
@@ -25,115 +27,136 @@ export async function callFnWhenAwait<T>(
   return t;
 }
 
-enum PromiseStatus {
-  resolved,
-  rejected,
-  pending,
-}
+export class PromiseWithStatus<T> implements Promise<T> {
```

Comment:
Can you add a jsdoc comment here?

Eg: PromiseWithStatus is used to handle promises with a status property rather than a try/catch or .then / .catch. This is achieved by suppressing promise rejections. For inspiration, see [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled)